### PR TITLE
Update guidance on named slots in framework components

### PR DESCRIPTION
**Pull request by natemoo-re**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [x] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- Follow-up to https://github.com/withastro/astro/pull/3652
- Adds guidance around using named slots with framework components

### CODE REVIEW

File: src/pages/en/core-concepts/framework-components.md [hunk 1]
```diff
@@ -119,32 +119,93 @@ import MyVueComponent from '../components/MyVueComponent.vue';
...
+  <header><slot name="title" /></header>
+  <main><slot /></main>
+  <footer><slot name="social-links" /></footer>
+</aside>
+```
+
 ## Nesting Framework Components
 
-Inside of an Astro component, you can also nest components from multiple frameworks.
+Inside of an Astro file, component children can also be hydrated components. This means that you can recursively nest components from any framework.
```
Comment (sarah11918) [inline src/pages/en/core-concepts/framework-components.md:182]:
```suggestion
Inside of an Astro file, framework component children can also be hydrated components. This means that you can recursively nest components from any of these frameworks.
```
2 suggestions to consider:

1. We've been saying "framework component" or "UI framework component" to distinguish from Astro component (which is... not a framework? That's another discussion!)

2. I also added "of these" to imply "of the UI frameworks we support" ... not ANY framework.

Comment (natemoo-re) [reply src/pages/en/core-concepts/framework-components.md]: Great suggestion, thank you!

**Comment by sarah11918** [comment]
Just one sentence commented above, otherwise, LGTM.

@Jutanium - care to provide a DINM? (Dan is not mad.)

**Comment by Jutanium** [comment]
Out of Astro time for today for a full review + comments. DINM, but instead of destructuring the props, please just accept `props` as the parameter and refer to `props.title`, `props.children` etc, because this works in _all_ the JSX frameworks (destructuring loses reactivity in Solid).

**Comment by sarah11918** [comment]
Update branch, to appease the Hippo checks, and LGTM! Thanks Nate! 💜

**Pull request merged**